### PR TITLE
bug/875_bad_split_on_http_responses

### PR DESCRIPTION
**Pull request by pcoello25**
* Fix issue #875 
* 100% unit test passed: 
```
Results : Tests run: 80, Failures: 0, Errors: 0, Skipped: 0
```
* e2e (unofficial) test (Testing on `OrionCKANSink` because uses `HttpBackend`) :
```
CKAN: Showing payloads. 
(organization_show -> organization_create -> package_create -> resource_create -> datastore_create -> resource_view_create -> datastore_upsert)

time=2016-03-16T11:28:05.711CET | lvl=DEBUG | trans=1458124024-407-0000000000 | srv=fiservice | subsrv=fiservicepath | function=createJsonResponse | comp=Cygnus | msg=com.telefonica.iot.cygnus.backends.http.HttpBackend[279] : response payload: {"help": "http://demo.ckan.org/api/3/action/help_show?name=organization_show", "success": false, "error": {"message": "Not found", "__type": "Not Found Error"}}
time=2016-03-16T11:28:06.381CET | lvl=DEBUG | trans=1458124024-407-0000000000 | srv=fiservice | subsrv=fiservicepath | function=createJsonResponse | comp=Cygnus | msg=com.telefonica.iot.cygnus.backends.http.HttpBackend[279] : response payload: {"help": "http://demo.ckan.org/api/3/action/help_show?name=organization_create", "success": true, "result": {"users": [{"email_hash": "xxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxx", "about": null, "capacity": "admin", "name": "user", "created": "2015-12-10T09:10:25.330843", "openid": null, "sysadmin": false, "activity_streams_email_notifications": false, "state": "active", "number_of_edits": 109, "display_name": "username", "fullname": "username", "id": "xxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxx", "number_created_packages": 5}], "display_name": "fiservice", "description": "", "image_display_url": "", "package_count": 0, "created": "2016-03-16T10:28:01.776123", "name": "fiservice", "is_organization": true, "state": "active", "extras": [], "image_url": "", "groups": [], "type": "organization", "title": "", "revision_id": "xxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxx", "num_followers": 0, "id": "xxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxx", "tags": [], "approval_status": "approved"}}
time=2016-03-16T11:28:07.709CET | lvl=DEBUG | trans=1458124024-407-0000000000 | srv=fiservice | subsrv=fiservicepath | function=createJsonResponse | comp=Cygnus | msg=com.telefonica.iot.cygnus.backends.http.HttpBackend[279] : response payload: {"help": "http://demo.ckan.org/api/3/action/help_show?name=package_create", "success": true, "result": {"license_title": null, "maintainer": null, "relationships_as_object": [], "private": false, "maintainer_email": null, "num_tags": 0, "id": "xxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxx", "metadata_created": "2016-03-16T10:28:02.269641", "metadata_modified": "2016-03-16T10:28:02.284050", "author": null, "author_email": null, "state": "active", "version": null, "creator_user_id": "xxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxx", "type": "dataset", "resources": [], "num_resources": 0, "tags": [], "groups": [], "license_id": null, "relationships_as_subject": [], "organization": {"description": "", "created": "2016-03-16T10:28:01.776123", "title": "", "name": "fiservice", "is_organization": true, "state": "active", "image_url": "", "revision_id": "xxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxx", "type": "organization", "id": "xxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxx", "approval_status": "approved"}, "name": "fiservice_fiservicepath", "isopen": false, "url": null, "notes": null, "owner_org": "xxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxx", "extras": [], "title": "fiservice_fiservicepath", "revision_id": "xxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxx"}}
time=2016-03-16T11:28:08.321CET | lvl=DEBUG | trans=1458124024-407-0000000000 | srv=fiservice | subsrv=fiservicepath | function=createJsonResponse | comp=Cygnus | msg=com.telefonica.iot.cygnus.backends.http.HttpBackend[279] : response payload: {"help": "http://demo.ckan.org/api/3/action/help_show?name=resource_create", "success": true, "result": {"cache_last_updated": null, "cache_url": null, "mimetype_inner": null, "hash": "", "description": "", "format": "", "url": "http://none", "created": "2016-03-16T10:28:03.718952", "state": "active", "webstore_last_updated": null, "webstore_url": null, "package_id": "xxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxx", "last_modified": null, "mimetype": null, "url_type": null, "position": 0, "revision_id": "xxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxx", "size": null, "id": "xxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxx", "resource_type": null, "name": "room1_room"}}
time=2016-03-16T11:28:08.874CET | lvl=DEBUG | trans=1458124024-407-0000000000 | srv=fiservice | subsrv=fiservicepath | function=createJsonResponse | comp=Cygnus | msg=com.telefonica.iot.cygnus.backends.http.HttpBackend[279] : response payload: {"help": "http://demo.ckan.org/api/3/action/help_show?name=datastore_create", "success": true, "result": {"fields": [{"type": "int", "id": "recvTimeTs"}, {"type": "timestamp", "id": "recvTime"}, {"type": "text", "id": "fiwareServicePath"}, {"type": "text", "id": "entityId"}, {"type": "text", "id": "entityType"}, {"type": "text", "id": "attrName"}, {"type": "text", "id": "attrType"}, {"type": "json", "id": "attrValue"}, {"type": "json", "id": "attrMd"}], "method": "insert", "resource_id": "xxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxx"}}
time=2016-03-16T11:28:09.353CET | lvl=DEBUG | trans=1458124024-407-0000000000 | srv=fiservice | subsrv=fiservicepath | function=createJsonResponse | comp=Cygnus | msg=com.telefonica.iot.cygnus.backends.http.HttpBackend[279] : response payload: {"help": "http://demo.ckan.org/api/3/action/help_show?name=resource_view_create", "success": true, "result": {"description": null, "title": "Recline grid view", "resource_id": "xxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxx", "view_type": "recline_grid_view", "id": "xxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxx", "package_id": "xxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxx"}}
time=2016-03-16T11:28:09.622CET | lvl=DEBUG | trans=1458124024-407-0000000000 | srv=fiservice | subsrv=fiservicepath | function=createJsonResponse | comp=Cygnus | msg=com.telefonica.iot.cygnus.backends.http.HttpBackend[279] : response payload: {"help": "http://demo.ckan.org/api/3/action/help_show?name=datastore_upsert", "success": true, "result": {"records": [{"attrType": "float", "recvTime": "2016-03-16T10:28:03.182Z", "recvTimeTs": "1458124083", "attrValue": "1.01", "entityType": "Room", "attrName": "temperature", "fiwareServicePath": "fiservicepath", "entityId": "Room1"}], "method": "insert", "resource_id": "xxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxx"}}
```
* Assignee @frbattid

### CODE REVIEW

File: src/main/java/com/telefonica/iot/cygnus/backends/http/HttpBackend.java [hunk 1]
```diff
@@ -269,7 +269,13 @@ private JsonResponse createJsonResponse(HttpResponse httpRes) throws Exception {
...
                 BufferedReader reader = new BufferedReader(new InputStreamReader(httpRes.getEntity().getContent()));
-                String res = reader.readLine();
+                String res = "";
+                String line;
+
+                while ((line = reader.readLine()) != null) {
+                    res += line;
+                } // while
+
                 LOGGER.debug("response payload: " + res);
```
Comment (frbattid) [inline src/main/java/com/telefonica/iot/cygnus/backends/http/HttpBackend.java]: It is not part of this issue, but we can do it right now: "Http response payload: "

Comment (pcoello25) [inline src/main/java/com/telefonica/iot/cygnus/backends/http/HttpBackend.java]: Fixed in 784a226

Comment (pcoello25) [inline src/main/java/com/telefonica/iot/cygnus/backends/http/HttpBackend.java]: Re-fixed in ae70ff5

**Comment by frbattid** [comment]
LGTM

**Pull request merged**